kong/router.lua

```diff
@@ -228,6 +230,12 @@ local function marshall_route(r)
 
         local wildcard_host_regex = host:gsub("%.", "\\.")
                                         :gsub("%*", ".+") .. "$"
+
+        _, host_port = utils.split_port(host)
+        if not host_port then
+          wildcard_host_regex = wildcard_host_regex:gsub('%$$', [[(:\d+)?$]])
```

Comment:
Better making this group a non-capturing one?

Reply:
done